diff --git a/pyproject.toml b/pyproject.toml
index 0179bff..77de13c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -71,7 +71,7 @@ OPENEXR_FORCE_INTERNAL_IMATH = 'ON'
 OPENEXR_FORCE_INTERNAL_OPENJPH = 'ON'
 OPENEXR_TEST_LIBRARIES = 'OFF'
 BUILD_SHARED_LIBS = 'OFF'
-CMAKE_OSX_DEPLOYMENT_TARGET = '10.15'
+CMAKE_OSX_DEPLOYMENT_TARGET = '11.0'
 CMAKE_POSITION_INDEPENDENT_CODE = 'ON'
 CMAKE_VERBOSE_MAKEFILE = 'ON'
 
@@ -88,7 +88,5 @@ manylinux-aarch64-image = "manylinux2014"
 
 # Needed for full C++17 support
 [tool.cibuildwheel.macos.environment]
-MACOSX_DEPLOYMENT_TARGET = "10.15"
-# pass compilers directly to avoid Homebrew LLVM libunwind
-CMAKE_ARGS = "-DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++"
+MACOSX_DEPLOYMENT_TARGET = "11.0"
 
